vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/CredentialsVerifier.kt

```diff
@@ -1,175 +1,28 @@
-            val signJWS: String = ldProofWithJWS.jws
-            val jwsObject: JWSObject = JWSObject.parse(signJWS)
-            val vcSignBytes: ByteArray = jwsObject.signature.decode()
-            val publicKeyJsonUri: URI = ldProofWithJWS.verificationMethod
-            val publicKeyObj = getPublicKeyFromVerificationMethod(publicKeyJsonUri)
-            if (Objects.isNull(publicKeyObj)) {
-                throw PublicKeyNotFoundException("Public key object is null")
-            }
-            val actualData: ByteArray = JWSUtil.getJwsSigningInput(jwsObject.header, canonicalHashBytes)
-            val jwsHeader: String = jwsObject.header.algorithm.name
-            verifyCredentialSignature(jwsHeader, publicKeyObj, actualData, vcSignBytes)
-        } catch (e: Exception) {
-            when (e) {
-                is PublicKeyNotFoundException,
-                is SignatureVerificationException -> throw e
-                else -> {
-                    throw UnknownException("Error while doing verification of verifiable credential")
-                }
-            }
+        if(credentials==null){
+            Log.e(tag, "Error - Input credential is null")
+            throw RuntimeException("Input credential is null")
         }
+        return CredentialVerifierFactory().verify(credentials,CredentialFormat.LDP_VC)
     }
 
-    @Throws(CertificateException::class, KeyStoreException::class, KeyManagementException::class)
-    private fun getPublicKeyFromVerificationMethod(publicKeyJsonUri: URI): PublicKey? {
-        return try {
-            val okHttpClient = OkHttpClient.Builder().build().newBuilder().build()
-            val request = Request.Builder()
-                .url(publicKeyJsonUri.toURL())
-                .get()
-                .build()
 
-            val response = okHttpClient.newCall(request).execute()
-            response.body?.let { responseBody ->
-                val objectMapper = ObjectMapper()
-                val jsonNode = objectMapper.readTree(responseBody.string())
-                if (jsonNode.isObject) {
-                    val responseObjectNode = jsonNode as ObjectNode
-                    val publicKeyPem =
-                        responseObjectNode[CredentialVerifierConstants.PUBLIC_KEY_PEM].asText()
-                    val strReader = StringReader(publicKeyPem)
-                    val pemReader = PemReader(strReader)
-                    val pemObject = pemReader.readPemObject()
-                    val pubKeyBytes = pemObject.content
-                    val pubKeySpec = X509EncodedKeySpec(pubKeyBytes)
-                    val keyFactory = KeyFactory.getInstance("RSA")
-                    keyFactory.generatePublic(pubKeySpec)
-                } else null
-            }
-        } catch (e: Exception) {
-            Log.e(tag, "Error Generating public key object", e)
-            null
-        }
+    fun verify(credential: String, credentialFormat: CredentialFormat): Boolean {
```

Comment:
If we are getting the whole VC with format and credential, why don't we pass that whole VC to the library as single param and extracting out the format in library itself ?

Reply:
After draft13 OpenId4VCI, VC response doesn't include format